### ISSUE TITLE
Abiotic Factor - Error: at offset 6211: unknown property type: ""

### ISSUE THREAD

**Issue by SirPneumono** [trumank/uesave-rs#51]
Error coming after a recent update when attempting to use uesave-rs on an Abiotic Factor .sav used for storing player data.

Attached player save file:

[Player_76561198984427717.sav.zip](https://github.com/user-attachments/files/17088399/Player_76561198984427717.sav.zip)

**Comment by trumank** [comment]
Abiotic Factor is now UE 5.4 which has a new implementation of property tag serialization not yet supported by this library.

**Comment by trumank** [comment]
I've implemented enough of the new tag serialization to parse this save. There are definitely still a lot of unhandled cases, so any more test saves would be appreciated. Progress is on the ue5.4 branch: https://github.com/trumank/uesave-rs/tree/ue5.4

**Comment by fang64** [comment]
> I've implemented enough of the new tag serialization to parse this save. There are definitely still a lot of unhandled cases, so any more test saves would be appreciated. Progress is on the ue5.4 branch: https://github.com/trumank/uesave-rs/tree/ue5.4

I just tried to use 
```cargo install --git https://github.com/trumank/uesave-rs --branch ue5.4```

it basically error'ed out:
```
error[E0425]: cannot find function `read_from_stream` in crate `ser_hex`
   --> uesave_cli/src/main.rs:144:26
    |
144 |                 ser_hex::read_from_stream("trace.json", &mut input, |reader| {
    |                          ^^^^^^^^^^^^^^^^ not found in `ser_hex`
```

**Comment by trumank** [comment]
Ah I need to fix dependency versioning. For now you can use --locked:

`cargo install --git https://github.com/trumank/uesave-rs --branch ue5.4`

**Comment by trumank** [comment]
I've refactored tags so they they are explicitly serialized to JSON instead of trying to shove all the necessary data into the properties themselves, which makes for a much cleaner implementation. I've also implemented Map/Set properties.

Consequence of tags being serialized to JSON is now some redundant (and useless) information is present (duplicate property name/index + raw size) and bool property values being present twice (once in the tag and once in the property data; only the tag actually matters).

I'll iron these out in another pass.
